source/structuring.rst

```diff
@@ -14,402 +14,468 @@ functions is better than copying-and-pasting duplicate bits of code
+   not create a named location and has a different behavior that is
+   covered elsewhere.
 
-This provides a unique identifier for the schema, as well as, in most
-cases, indicating where it may be downloaded.
+.. index::
+   single: $ref
+   single: structuring; $ref
 
-But be aware of the second purpose of the ``$id`` property: that it
-declares a base URI for ``$ref`` URI-references elsewhere in the file.
-For example, if you had:
+.. _ref:
 
-.. schema_example::
+$ref
+----
 
-  { "$ref": "person.json" }
+We can now use our knowledge of how schemas are identified, to
+reference them in our schemas using the ``$ref`` keyword. The value
+of ``$ref`` is a relative or absolute URI that is resolved against the
+schema's identifier. When a reference is encountered, the schema its
+value identifies is retrieved and execution is continued from that
+point.
 
-in the same file, a JSON schema validation library that supported network
-fetching may fetch ``person.json`` from
-``http://foo.bar/schemas/person.json``, even if ``address.json`` was loaded from
-somewhere else, such as the local filesystem. The drafts do not define this
-area of behaviour very clearly, and validator implementations may vary in
-exactly how they try to locate the referenced schema.
+``$ref`` can be used anywhere a schema is expected. When an object
+contains a ``$ref`` property, the object is considered a reference,
+not a schema. Therefore, any other properties you put there will not
+be treated as JSON Schema keywords and be ignored by the validator.
 
+For this example, let's say we want to define a customer record, where
+each customer may have both a shipping and a billing address.
+Addresses are always the same---they have a street address, city and
+state---so we don't want to duplicate that part of the schema
+everywhere we want to store an address.  Not only would that make the
+schema more verbose, but it makes updating it in the future more
+difficult.  If our imaginary company were to start doing international
+business in the future and we wanted to add a country field to all the
+addresses, it would be better to do this in a single place rather than
+everywhere that addresses are used.
 
-|draft6|
+.. schema_example::
 
-.. draft_specific::
+    {
+      "$id": "https://example.com/schemas/customer",
 
-    --Draft 4
-    In Draft 4, ``$id`` is just ``id`` (without the dollar sign).
+      "type": "object",
+      "properties": {
+        "first_name": { "type": "string" },
+        "last_name": { "type": "string" },
+        "shipping_address": { "$ref": "/schemas/address" },
+        "billing_address": { "$ref": "/schemas/address" }
+      },
+      "required": ["first_name", "last_name", "shipping_address", "billing_address"]
+    }
 
-The ``$id`` property should never be the empty string or an empty fragment
-(``#``), since that doesn't really make sense.
+The relative URIs in ``$ref`` resolves against the schema's identifier
```

Comment:
1) The relative URIs [...] *resolve_* against...
2) The example before uses `/schemas/address`. Wouldn't that be *absolute paths* (rooted in the URI domain) whereas *relative paths* would always begin with a `./` or `../`?

Reply:
I'm using the definitions in [RFC3986 Section 4.1](https://datatracker.ietf.org/doc/html/rfc3986#section-4.1), 4.2, and 4.3. The thing that makes these URIs relative is the lack of a scheme and domain. At least a link to the RFC is probably warranted if not a short section near the top defining terminology.

Reply:
Always good to have RFCs to talk about. :+1 Indeed I thought about an URIs `path` segment, only, without noticing, though, that leading 'dots' don't matter to the definition of *relative URI-reference*. But using the term *relative URI-Reference* from the RFC here, would probably not have helped me, either, so I don't ask for changing the wording here. :smiley:  A quote may still make sense. 

Just a minor side node (warning I may be a bit nitpick, here): as far as I understood section 4 *lack of a scheme* is sufficient to make an URI a relative URI-reference. Since the ABNF for an URI-Reference contains *authority* it appears as if it could still contain a *domain* (given you used *domain* interchangibly for *authority*) . See [Appendix B](https://datatracker.ietf.org/doc/html/rfc3986#appendix-B) and `$4`.